### PR TITLE
update log plot and add image plot

### PR DESCRIPTION
**Pull request by jzstark**
Sorry for this late PR. Here is what it contains:

1. in the new version, `semilogx` and `semilogy` are only special cases of `loglog`, to which I  added an `axis` parameter.
2. Add a image plot. Please see here ([code](https://gist.githubusercontent.com/jzstark/9f49c77626533fb8e84a5131f67d4e05/raw/80da435d38312de63bb841a5af61853c88571775/examples.ml), [figure](https://gist.githubusercontent.com/jzstark/9f49c77626533fb8e84a5131f67d4e05/raw/80da435d38312de63bb841a5af61853c88571775/plot_026.png)) for an example. To remove the ticks/labels around the image, I add a `is_image` property to `page` in order for `_config_2d_axis` to return the proper code -1.

### CODE REVIEW

File: lib/owl_plot.ml [hunk 1]
```diff
@@ -1560,6 +1578,38 @@ let scatterhist = None
...
+let image ?(h=_default_handle) mat=
+  let open Plplot in
+  (* compute necessary parameters *)
+  let width, height = Owl_dense_matrix.D.shape mat in
+  let num_col = Owl_dense_matrix.D.max mat in
+  let img = Owl_dense_matrix.D.to_arrays mat in
+  let width = float_of_int width in
+  let height = float_of_int height in
+  let num_col = int_of_float num_col in
+  (* specify the boundary of imageplot*)
```
Comment (ryanrhymes) [inline lib/owl_plot.ml]: use consistent format (* ... *)

**Comment by ryanrhymes** [comment]
Thanks, this is really useful PR.  A couple of suggestions: 1) we perhaps need to change the function name `image` to something else if it only shows grayscale image; 2) `image` can be used to showcase the neural network examples in future; 3) I think `axis` can be part of the `spec`, so I will refactor the code a bit later, remember to `fetch upstream` to sync :)

Nice PR and great job! 👍

**Comment by hcarty** [comment]
It may be worth using `plimagefr` for this as it would allow more control over the color range used.

**Comment by jzstark** [comment]
@hcarty  Thanks for your suggestion. Currently I'm trying to use `plplot` to draw color images. I find functions that can manipulate/load colormap palette and map color index to continuous colors, but not a function that can map RGB color back to color index, since my understanding is that `plimage` and `plimagefr` only accept one index color matrices as input, while the image data I have consist of 3 matrices, one for each color channel. I've checked related documents but must have missed something. It will be really appreciated if any suggestion about it could be kindly provided. Thanks!

**Comment by hcarty** [comment]
@jzstark plplot does not have a function to map a specific color to a specific matrix element.  `plimagefr` gives more control over the color range used within a continuous color palette but it doesn't allow directly mapping colors.

It's something that would be good to have in plplot in general so if you want to get your hands dirty in some C then I suspect that a contribution would be welcome!  I haven't been actively involved in the project for a while though unfortunately so I can't offer much direct insight.

**Pull request merged**